### PR TITLE
feat: openapi export

### PR DESCRIPTION
**Pull request by mallachari**
Fixes: https://github.com/stoplightio/elements/issues/707
Requires https://github.com/stoplightio/platform-internal/pull/7315 to make export button visible in StoplightProject (it's on v2 but not yet in prod)

Adds export/download feature to API and StopolightProject components.

Screenshot 2021-07-28 at 16 04 42
Screenshot 2021-07-28 at 16 04 51

### CODE REVIEW

**Comment by marcelltoth** [comment]
I'd make the button more neutral, maybe outlined, as in platform. But @mnaumanali94 will decide

**Review by mpodlasin — APPROVED** [review]
I agree with @marcelltoth 

However all the rest looks and works good.

**Comment by mallachari** [comment]
`minimal` appearance doesn't look good but `default` does. Switched to:
Screenshot 2021-07-28 at 22 19 34

File: packages/elements/src/containers/API.tsx [hunk 1]
```diff
@@ -90,10 +98,44 @@ export const APIImpl: React.FC<APIProps> = props => {
...
+    const originalDocument = typeof document === 'object' ? JSON.stringify(document, null, 2) : document || '';
+    exportDocument(originalDocument);
+  }, [document, exportDocument]);
+
+  const exportBundledDocument = React.useCallback(() => {
+    const stringifiedDocument = isJsonDocument
+      ? JSON.stringify(bundledDocument, null, 2)
+      : safeStringify(bundledDocument);
+    exportDocument(stringifiedDocument);
+  }, [bundledDocument, isJsonDocument, exportDocument]);
```
Comment (marcelltoth) [inline packages/elements/src/containers/API.tsx:128]: WDYT about extracting these into a custom hook, to clean up the container?

File: packages/elements/src/utils/oas/index.ts [hunk 2]
```diff
@@ -146,3 +146,12 @@ function findMapMatch(key: string | number, map: ISourceNodeMap[]): ISourceNodeM
     }
   }
 }
+
+export function isJson(value: string) {
+  try {
+    JSON.parse(value);
+  } catch (e) {
+    return false;
+  }
```
Comment (marcelltoth) [inline packages/elements/src/utils/oas/index.ts:155]: Lol, I like this one 😄

**Review by marcelltoth — DISMISSED** [review]
Actually, can you write some tests for this? It's a significant piece of functionality (`hideExport`, handling json / yaml), we should cover it.

File: packages/elements-core/src/components/Docs/HttpService/HttpService.spec.tsx [hunk 3]
```diff
@@ -241,4 +242,56 @@ describe('HttpService', () => {
...
+            />
+          </MosaicProvider>
+        </Router>,
+      );
+
+      const exportButton = wrapper.queryByRole('button', { name: 'Export' });
+      expect(exportButton).not.toBeInTheDocument();
+    });
+
+    it('should not render if hno exportProps are present', () => {
```
Comment (mpodlasin) [inline packages/elements-core/src/components/Docs/HttpService/HttpService.spec.tsx:284]: typo

**Review by mpodlasin — APPROVED** [review]
👍

**Comment by cancan101** [comment]
Is there any documentation on what exporting "Bundled References" means (as opposed to a exporting "Original")?

**Comment by mnaumanali94** [comment]
@cancan101 While there is no documentation on this original export doesn't resolve references while bundled resolves references and places them in the components section so that there is no duplication. Generally bundled export would be your best bet.

**Comment by mnaumanali94** [comment]
@pamgoodrich probably a good topic to document at some point

**Pull request merged**